src/game.cpp

```diff
@@ -848,6 +851,68 @@ bool game::start_game()
     return true;
 }
 
+void game::place_starting_vehicle( const cata::optional<vproto_id> &starting_vehicle )
+{
+    vehicle veh( *starting_vehicle );
+    std::vector<std::string> omt_search_types;
+    if( veh.can_float() ) {
+        omt_search_types.push_back( "river" );
+        omt_search_types.push_back( "lake" );
+    }
+    if( veh.max_ground_velocity() > 0 ) {
+        omt_search_types.push_back( "road" );
+        omt_search_types.push_back( "field" );
+    }
+    if( omt_search_types.empty() ) {
+        return;
+    }
+    find_location_for_starting_vehicle( *starting_vehicle, omt_search_types );
+
+}
+
+void game::find_location_for_starting_vehicle( const vproto_id id,
+        std::vector<std::string> &omt_search_types )
+{
+    for( const std::string &search_type : omt_search_types ) {
+        omt_find_params find_params;
+        find_params.must_see = false;
+        find_params.cant_see = false;
+        find_params.types.emplace_back( search_type, ot_match_type::type );
+        for( int count = 0; count < 10; ++count ) {
```

Comment:
I suppose you don't need this loop anymore: after #37482, `find_all` will return points sorted by proximity.